code/modules/paperwork/photography.dm

```diff
@@ -434,3 +439,36 @@
 	C.toner -= 20	 //Cyborgs are very ineffeicient at printing an image
 	visible_message("[C.name] spits out a photograph from a narrow slot on it's chassis.")
 	usr << "<span class='notice'>You print a photograph.</span>"
+
+/obj/item/device/camera/spiritcam/attack_self(mob/user)
+	if(istype(user, /mob/living) && user.mind)
+		if(user.mind.special_role && on)
+			if(remote)
+				user << "<span class='notice'>You set the camera back to normal.</span>"
+				remote = 0
+			else
+				user << "<span class='notice'>You push a hidden button and activate spirit photography.</span>"
+				remote = 1
+
+/obj/item/device/camera/spiritcam/afterattack(atom/target, mob/user, flag)
+	if(remote && on && pictures_left)
+		var/list/real_living_players = list()
+		var/list/wanted_players = list()
+		var/wanted = copytext(sanitize(input(user, "Who would you like to take a photograph of?", "Target name")as text | null),1,26)
+		for(var/mob/M in living_mob_list)
+			real_living_players += M.real_name
+			real_living_players[M.real_name] = M
+			if(M.real_name == wanted)
+				wanted_players += M
+				wanted_players[M] = M
```

Comment:
nigga you don't need this line at all, why are you making wanted_players an associative list,you just want the actual mobs inside this list no?